src/native/libs/System.Globalization.Native/CMakeLists.txt

```diff
@@ -97,6 +112,9 @@ if (GEN_SHARED_LIB)
         include(CMakeFindFrameworks)
         find_library(FOUNDATION Foundation REQUIRED)
     endif()
+    if (CLR_CMAKE_TARGET_MACCATALYST OR CLR_CMAKE_TARGET_IOS OR CLR_CMAKE_TARGET_TVOS)
+        find_library(ICUCORE icucore REQUIRED)
```

Comment:
not sure where this will actually find icucore from. I think I'd prefer turning off GEN_SHARED_LIB for iOS/Catalyst etc in general for System.Globalization.Native, it wouldn't do the right thing in the non-static case.

Reply:
or we need to make a .dylib of the "static" version where it links in the icu .a

Reply:
This is actually a leftover that I forgot to remove. The branch is never taken on the iOS-like platforms.